src/ducks/apps/index.js

```diff
@@ -316,48 +316,45 @@ export function fetchLatestApp(slug, channel = DEFAULT_CHANNEL) {
   }
 }
 
-export function getFormattedRegistryApp(response, channel) {
-  return cozy.client
-    .fetchJSON('GET', `/registry/${response.slug}/${channel}/latest`)
-    .then(version => {
-      // FIXME retro-compatibility for old formatted manifest
-      const manifest = _sanitizeOldManifest(version.manifest)
-      // source is only used by the stack when installed
-      // so we removed it from the app manifest if present
-      if (manifest.source) delete manifest.source
-
-      const versionFromRegistry = version.version
-      const screensLinks =
-        manifest.screenshots &&
-        manifest.screenshots.map(name => {
-          const fileName = name.replace(/^.*[\\/]/, '')
-          return `${cozy.client._url}/registry/${
-            manifest.slug
-          }/${versionFromRegistry}/screenshots/${fileName}`
-        })
-      const iconLink = `/registry/${manifest.slug}/${versionFromRegistry}/icon`
-      return _getIcon(iconLink).then(iconData => {
-        return Object.assign(
-          {},
-          {
-            versions: response.versions
-          },
-          manifest,
-          {
-            icon: iconData,
-            version: versionFromRegistry,
-            type: version.type,
-            categories: _sanitizeCategories(manifest.categories),
-            uninstallable: !config.notRemovableApps.includes(manifest.slug),
-            isInRegistry: true,
-            // add screensLinks property only if it exists
-            ...(screensLinks ? { screenshots: screensLinks } : {}),
-            // add installed value only if not already provided
-            installed: response.installed || false
-          }
-        )
-      })
+export function getFormattedRegistryApp(responseApp) {
+  const version = responseApp.latest_version
+  // FIXME retro-compatibility for old formatted manifest
+  const manifest = _sanitizeOldManifest(version.manifest)
+  // source is only used by the stack when installed
+  // so we removed it from the app manifest if present
+  if (manifest.source) delete manifest.source
```

Comment:
Do you think we could put this part inside `_sanitizeOldManifest()` ?

Reply:
Finally nope, we can't because here it's only from registry but this helper is also use when we get apps from the stack. And the source in manifest is used by the stack differently, when the application is installed this source is used to get the channel.
To be clearer, the source should not be provided by the registry in the manifest. If it is, it will replaced by the one used by the stack in all cases.